spark/src/test/scala/org/apache/spark/sql/delta/rowtracking/RowTrackingConflictResolutionSuite.scala

```diff
@@ -147,4 +157,172 @@ class RowTrackingConflictResolutionSuite extends QueryTest
+      val (file1, file2, file3, file4) = setupTableAndGetAllFiles(log)
+
+      // Start Backfill.
+      val backfillTxn = log.startTransaction()
+
+      // A delete occurs in parallel. Delete wins.
+      val deleteTxn = log.startTransaction()
+      deleteTxn.filterFiles(EqualTo('x, Literal(1)) :: Nil)
+      val deleteActions = Seq(file1.remove, file2.remove)
+      // Truncate is a data-changing operation.
+      deleteTxn.commit(deleteActions, Truncate())
+
+      // Finish backfill.
+      executeBackfill(log, backfillTxn)
+
+      val finalFiles = log.update().allFiles.collect()
+      finalFiles.foreach(a => assert(a.baseRowId.nonEmpty))
+      assertRowIdsAreValid(log)
+      assert(finalFiles.map(_.path).toSet === Seq(file3, file4).map(_.path).toSet)
+    }
+  }
+
+  test("Backfill conflicts with a delete, Backfill wins") {
+    withTempDir { dir =>
+      val log = DeltaLog.forTable(spark, dir.getCanonicalPath)
+      // Setup
+      val (file1, file2, file3, file4) = setupTableAndGetAllFiles(log)
+
+      // Start delete
+      val deleteTxn = log.startTransaction()
+      deleteTxn.filterFiles(EqualTo('x, Literal(1)) :: Nil)
+
+      // Backfill occurs in parallel and wins.
+      val backfillTxn = log.startTransaction()
+      executeBackfill(log, backfillTxn)
+
+      val deleteActions = Seq(file1.remove, file2.remove)
+      // Truncate is a data-changing operation.
+      deleteTxn.commit(deleteActions, Truncate())
+
+      val finalFiles = log.update().allFiles.collect()
+      finalFiles.foreach(a => assert(a.baseRowId.nonEmpty))
+      assertRowIdsAreValid(log)
+      assert(finalFiles.map(_.path).toSet === Seq(file3, file4).map(_.path).toSet)
+    }
+  }
+
+  test("Backfill conflicts with a DV delete, Delete wins") {
+    withTempDir { dir =>
+      val log = DeltaLog.forTable(spark, dir.getCanonicalPath)
+
+      // Setup
+      val (file1, file2, file3, file4) = setupTableAndGetAllFiles(log)
+      enableDeletionVectorsInTable(log)
+
+      // Start Backfill
+      val backfillTxn = log.startTransaction()
+
+      // A delete occurs in parallel. Delete wins.
+      val deleteTxn = log.startTransaction()
+      deleteTxn.filterFiles(EqualTo('x, Literal(1)) :: Nil)
+      val (addFile1WithDV, removeFile1) = addDVToFileInTable(log, file1)
+      val (addFile2WithDV, removeFile2) = addDVToFileInTable(log, file2)
+      val deleteActions = Seq(addFile1WithDV, removeFile1, addFile2WithDV, removeFile2)
+      // Truncate is a data-changing operation.
+      deleteTxn.commit(deleteActions, Truncate())
+
+      // Finish Backfill
+      executeBackfill(log, backfillTxn)
+
+      val finalFiles = log.update().allFiles.collect()
+      finalFiles.foreach(a => assert(a.baseRowId.nonEmpty))
+      assertRowIdsAreValid(log)
+      val allFiles = Seq(file1, file2, file3, file4)
+      assert(finalFiles.map(_.path).toSet === allFiles.map(_.path).toSet)
+    }
+  }
+
+  test("Backfill conflicts with a DV delete, Backfill wins") {
```

Comment:
The PR description mentions you'll add more tests later so I'll be patient, but it would be interesting to test other conflict cases where the other transaction isn't a plain delete

Reply:
@johanl-db Yes, this requires extensive testing so this PR will be huge if we try to add them all here, while Row Tracking Backfill is not enabled we can add them one by one.